libshaderc/src/shaderc_cpp_test.cc

```diff
@@ -71,6 +71,91 @@ const char kTwoWarningsShader[] =
     "attribute float y;\n"
     "void main(){}\n";
 
+// Empty 310 es shader. It is valid for vertex, fragment, compute shader kind.
+const char kEmpty310ESShader[] =
+    "#version 310 es\n"
+    "void main() {}\n";
+
+// Vertex only shader.
+const char kVertexOnlyShader[] =
+    "#version 310 es\n"
+    "void main() {\n"
+    "    gl_Position = vec4(1.);\n"
+    "}";
+
+// TessControl only shader.
+const char kTessControlOnlyShader[] =
+    "#version 440 core\n"
+    "layout(vertices = 3) out;\n"
+    "void main() { }";
+
+// TessEvaluation only shader.
+const char kTessEvaluationOnlyShader[] =
+    "#version 440 core\n"
+    "layout(triangles) in;\n"
+    "void main() { }";
+
+// Geometry only shader.
+const char kGeometryOnlyShader[] =
+    "#version 150 core\n"
+    "layout (triangles) in;\n"
+    "layout (line_strip, max_vertices = 4) out;\n"
+    "void main() { }";
+
+// Vertex only shader with #pragma annotation.
+const char kVertexOnlyShaderWithPragma[] =
+    "#version 310 es\n"
+    "#pragma shader_stage(vertex)\n"
+    "void main() {\n"
+    "    gl_Position = vec4(1.);\n"
+    "}";
+
+// Fragment only shader with #pragma annotation.
+const char kFragmentOnlyShaderWithPragma[] =
+    "#version 310 es\n"
+    "#pragma shader_stage(fragment)\n"
+    "void main() {\n"
+    "    gl_FragDepth = 10.;\n"
+    "}";
+
+// TessControl only shader with #pragma annotation.
+const char kTessControlOnlyShaderWithPragma[] =
+    "#version 440 core\n"
+    "#pragma shader_stage(tesscontrol)\n"
+    "layout(vertices = 3) out;\n"
+    "void main() { }";
+
+// TessEvaluation only shader with #pragma annotation.
+const char kTessEvaluationOnlyShaderWithPragma[] =
+    "#version 440 core\n"
+    "#pragma shader_stage(tesseval)\n"
+    "layout(triangles) in;\n"
+    "void main() { }";
+
+// Geometry only shader with #pragma annotation.
+const char kGeometryOnlyShaderWithPragma[] =
+    "#version 150 core\n"
+    "#pragma shader_stage(geometry)\n"
+    "layout (triangles) in;\n"
+    "layout (line_strip, max_vertices = 4) out;\n"
+    "void main() { }";
+
+// Compute only shader with #pragma annotation.
+const char kComputeOnlyShaderWithPragma[] =
+    "#version 310 es\n"
+    "#pragma shader_stage(compute)\n"
+    "void main() {\n"
+    "    uvec3 temp = gl_WorkGroupID;\n"
+    "}";
+
+// Vertex only shader with invalid #pragma annotation.
+const char kVertexOnlyShaderWithInvalidPragma[] =
+    "#version 310 es\n"
+    "#pragma shader_stage(fragment)\n"
+    "void main() {\n"
+    "    gl_Position = vec4(1.);\n"
+    "}";
+
```

Comment:
Duplicating them keeps it simple and isolated, less indirection.  
Coder's choice.  
But also, if you do commonize the data, avoid static-duration class objects.  Only POD allowed.

Others might disagree whether DRY would be better for this.